src/rgw/rgw_bucket.cc

```diff
@@ -653,6 +653,8 @@ int RGWBucket::link(RGWBucketAdminOpState& op_state, optional_yield y,
     return -EINVAL;
   }
   rgw_bucket old_bucket = bucket;
+  rgw_user user_id = op_state.get_user_id();
+  bucket.tenant = user_id.tenant;
```

Comment:
thanks! it looks like i messed this up in the metadata refactoring rebase here: https://github.com/ceph/ceph/pull/29118#discussion_r316867763